tekton/release-pipeline.yaml

```diff
@@ -8,139 +8,205 @@ spec:
+    - name: release-file-no-tag
+      description: the URL of the release file
+      value: $(tasks.report-bucket.results.release-no-tag)
   tasks:
+    - name: git-clone
+      taskRef:
+        name: git-clone
+      workspaces:
+        - name: output
+          workspace: workarea
+          subpath: git
+      params:
+        - name: url
+          value: https://$(params.package)
+        - name: revision
+          value: $(params.gitRevision)
     - name: precheck
+      runAfter: [git-clone]
       taskRef:
         name: prerelease-checks
       params:
         - name: package
           value: $(params.package)
         - name: versionTag
           value: $(params.versionTag)
-      resources:
-        inputs:
-          - name: release-bucket
-            resource: bucket
-          - name: source-to-release
-            resource: source-repo
+        - name: releaseBucket
+          value: $(params.releaseBucket)
+      workspaces:
+        - name: source-to-release
+          workspace: workarea
+          subpath: git
     - name: unit-tests
       runAfter: [precheck]
       taskRef:
-        name: golang-test-legacy
+        name: golang-test
       params:
         - name: package
           value: $(params.package)
         - name: flags
           value: -v -mod=vendor
-      resources:
-        inputs:
-          - name: source
-            resource: source-repo
+      workspaces:
+        - name: source
+          workspace: workarea
+          subpath: git
     - name: build
       runAfter: [precheck]
       taskRef:
-        name: golang-build-legacy
+        name: golang-build
       params:
         - name: package
           value: $(params.package)
-        - name: flags
-          value: -mod=vendor
-      resources:
-        inputs:
-          - name: source
-            resource: source-repo
+        - name: packages
+          value: ./cmd/...
+      workspaces:
+        - name: source
+          workspace: workarea
+          subpath: git
     - name: build-base-image
       runAfter: [build, unit-tests]
       taskRef:
         name: build-multiarch-base-image
       params:
-        - name: pathToProject
+        - name: package
           value: $(params.package)
         - name: imageRegistry
           value: $(params.imageRegistry)
+        - name: imageRegistryPath
+          value: $(params.imageRegistryPath)
         - name: platforms
           value: $(params.platforms)
-      resources:
-        inputs:
-          - name: source
-            resource: source-repo
-        outputs:
-          - name: builtBaseImage
-            resource: builtBaseImage
+        - name: serviceAccountPath
+          value: $(params.serviceAccountPath)
+      workspaces:
+        - name: source
+          workspace: workarea
+          subpath: git
+        - name: release-secret
+          workspace: release-secret
     - name: publish-images
-      runAfter: [build, unit-tests]
```

Comment:
This is interesting, executing `publish-image` after `build` and `unit-test` are done and in parallel with `build-base-image`. `publish-image` is not dependent on `build-base-image` through `resources` as well (or am I missing it? 😜 ). Wondering how did this work until now?